dataplane/fragmentation.cpp

```diff
@@ -1,19 +1,39 @@
 #include "fragmentation.h"
 #include "common.h"
 #include "controlplane.h"
-#include "dataplane.h"
+#include "metadata.h"
 
-fragmentation_t::fragmentation_t(cControlPlane* controlPlane,
-                                 cDataPlane* dataPlane) :
-        controlPlane(controlPlane),
-        dataPlane(dataPlane)
+namespace fragmentation
 {
-	memset(&stats, 0, sizeof(stats));
+
+Fragmentation::Fragmentation(OnCollected callback) :
+        callback_{callback}
+{}
+
+Fragmentation::Fragmentation(OnCollected callback, const FragmentationConfig& cfg) :
+        callback_{callback},
+        config_{cfg}
+{
+	memset(&stats_, 0, sizeof(stats_));
+}
+
+Fragmentation::Fragmentation(Fragmentation&& other)
+{
+	*this = std::move(other);
+}
```

Comment:
Same problem as in the first commit, it's better to use
```cpp
Fragmentation::Fragmentation(Fragmentation&& other) noexcept
    : callback_(std::move(other.callback_)),
      config_(std::move(other.config_)),
      stats_(std::move(other.stats_)),
      fragments_(std::move(other.fragments_)) {}
  ```

Reply:
Same here, duplicationg logic is IMHO more error prone